sql/core/src/test/scala/org/apache/spark/sql/StatisticsCollectionSuite.scala

```diff
@@ -650,4 +650,44 @@ class StatisticsCollectionSuite extends StatisticsCollectionTestBase with Shared
       }
     }
   }
+
+  test("Data source tables support fallback to HDFS for size estimation") {
+    // Non-partitioned table
+    withTempDir { dir =>
+      Seq(false, true).foreach { fallBackToHDFSForStats =>
+        withSQLConf(SQLConf.ENABLE_FALL_BACK_TO_HDFS_FOR_STATS.key -> s"$fallBackToHDFSForStats") {
```

Comment:
why this config has no effect in this test?

Reply:
// fallBackToHDFSForStats = true: The table stats will be recalculated by DetermineTableStats
// fallBackToHDFSForStats = false: The table stats will be recalculated by [FileIndex](https://github.com/apache/spark/blob/b276788d57b270d455ef6a7c5ed6cf8a74885dde/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/PartitioningAwareFileIndex.scala#L103)